src/main/java/org/tikv/common/PDClient.java

```diff
@@ -811,43 +819,60 @@ public String toString() {
     }
   }
 
-  private Metapb.Region decodeRegion(Metapb.Region region) {
+  @Override
+  public Metapb.Region decodeRegion(Metapb.Region region) {
```

Comment:
From line 367, it seems like we only don't need to do anything when API is v1, but why do we need decodeRegion now when API is v1. Is there a bug in old code?

Reply:
So we need a fast path to return when v1

Reply:
Has changed to using `RequestKeyCodec` to do this.